compiler/ccgtypes.nim

```diff
@@ -1527,8 +1566,14 @@ proc genHook(m: BModule; t: PType; info: TLineInfo; op: TTypeAttachedOp; result:
       localError(m.config, info,
         theProc.name.s & " needs to have the 'nimcall' calling convention")
 
-    genProc(m, theProc)
-    result.add theProc.loc.r
+    if op == attachedDestructor and t.kind == tySequence:
```

Comment:
```suggestion
    if op == attachedDestructor:
```

Reply:
Not every type supports T destructors. I suppose that's going to cause problems. It should be fine for `ref string` and `ref object`.

Reply:
That only needs you need to do some introspection and see if the destructor takes a `var T` or not. But special casing `tySequence` is clearly wrong.